sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -494,44 +494,54 @@ object DateTimeUtils {
   }
 
   /**
-   * Returns the month value for the given date. The date is expressed in days
-   * since 1.1.1970. January is month 1.
+   * Split date (expressed in days since 1.1.1970) into four fields:
+   * year, month (Jan is Month 1), dayInMonth, daysToMonthEnd (0 if it's last day of month).
    */
-  def getMonth(date: Int): Int = {
+  def splitDate(date: Int): (Int, Int, Int, Int) = {
```

Comment:
Could you also post a benchmark to compare with using java.util.Date? Just want to make sure we're doing the right things, also address the concern from @srowen , thanks!